dch/mqtt-subscriber/mqtt-subscriber.js

```diff
@@ -55,28 +76,24 @@ sysClient.on(MQTT_CONNECT_EVENT, function () {
 // Acceleration data is received here and is plaved into MongoDB
 dcappClient.on(MQTT_MESSAGE_EVENT, function (topic, message) {
   // Print for debugging 
-  console.log(message.toString());
-  // Connect to the db and insert received data
-  MongoClient.connect(
-    MONGODB_URL,
-    function(err, db) {
-      if(err) { return console.dir(err); }
-      var messageString = message.toString();
-      try {
-        var messageJson = JSON.parse(messageString);
-        var collection = db.collection(MQTT_COLLECTION);
-        var input = {
-          WATCH_ID: messageJson.watch_id,
-          TIMESTAMP: messageJson.timestamp,
-          X_ACCELERATION: messageJson.acc_x,
-          Y_ACCELERATION: messageJson.acc_y,
-          Z_ACCELERATION: messageJson.acc_z
-        };
-        collection.insert(input);
-      } catch (exception) {
-        console.log('Insert exception: ' + exception);
+  console.log("MQTT message: "+message.toString());
+  var decrypted = decipher.decryptText(message.toString());
+  var dataObject = getDataObject(decrypted);
+  if (dataObject){
+    // TODO should consider bulk insert
+    // TODO add gradient field
+    var data = Data(decrypted);
+    Data.save(message.toString(), function(err){
```

Comment:
It doesn't seem that the save method is meant to be used this way and we don't want to save undecrypted data. We can indicate in a clearer way if we want to use undecrypted data for now. Gonna change this as I add gradient.